File: 1erdepartamental/python/18.quick_sort/main.py
"""
Programa: Quick Sort
Demuestra algoritmo de ordenamiento quick sort
"""
#Clase principal
from core.ordenador import OrdenadorQuick

#Funcion Principal
def main ():
    numeros = [64, 34, 25, 12, 22, 11, 90] #Lista de numeros a acomodar

    print(f"Array original: {numeros}")

    ordenador = OrdenadorQuick()
    ordenador.ordenar(numeros)

    print(f"Array ordenado: {numeros}")
    print(f"Particiones: {ordenador.particiones}")

#Punto de Entrada
if __name__ == "__main__":
    main()

File: 1erdepartamental/c++/18.quick_sort/main.cpp
#include <iostream>
#include "operations/OrdenadorQuick.h"

int main() {
    std::vector<int> numeros = {64, 34, 25, 12, 22, 11, 90};
    
    std::cout << "Array original: ";
    for (int n : numeros) std::cout << n << " ";
    std::cout << std::endl << std::endl;
    
    OrdenadorQuick<int> ordenador;
    ordenador.ordenar(numeros);
    
    std::cout << "Array ordenado: ";
    for (int n : numeros) std::cout << n << " ";
    std::cout << std::endl << std::endl;
    
    std::cout << "Particiones: " << ordenador.getParticiones() << std::endl;
    
    return 0;
}
